obspy/taup/ray_paths.py

```diff
@@ -0,0 +1,137 @@
+                warnings.warn(msg)
+                continue
+            stlats.append(station.latitude)
+            stlons.append(station.longitude)
+            stlabels.append(label_)
+
+    # make a big list of event coordinates and names
+    # this part should be included as a subroutine of catalog that extracts
+    # a list of event properties. E.g. catalog.extract(['latitiude',
+    # 'longitude', 'depth', 'mag', 'focal_mechanism') The same should be done
+    # for an inventory with stations
+    evlats = []
+    evlons = []
+    evdepths = []
+    evlabels = []
+    for event in catalog:
+        if not event.origins:
+            msg = ("Event '%s' does not have an origin and will not be "
+                   "plotted." % str(event.resource_id))
+            warnings.warn(msg)
+            continue
+        if not event.magnitudes:
+            msg = ("Event '%s' does not have a magnitude and will not be "
+                   "plotted." % str(event.resource_id))
+            warnings.warn(msg)
+            continue
+        origin = event.preferred_origin() or event.origins[0]
+        evlats.append(origin.latitude)
+        evlons.append(origin.longitude)
+        if not origin.get('depth'):
+            origin.depth = 0.
+        evdepths.append(origin.get('depth') * 1e-3)
+        magnitude = event.preferred_magnitude() or event.magnitudes[0]
+        mag = magnitude.mag
+        label = '  {:s} | M{:.1f}'.format(str(origin.time.date), mag)
+        evlabels.append(label)
+
+    # initialize taup model if it is not provided
+    if isinstance(taup_model, str):
+        from obspy.taup import TauPyModel
+        model = TauPyModel(model=taup_model)
+    else:
+        model = taup_model
+
+    # now loop through all stations and source combinations
+    r_earth = model.model.radius_of_planet
+    greatcircles = []
+    for stlat, stlon, stlabel in zip(stlats, stlons, stlabels):
+        for evlat, evlon, evdepth_km, evlabel in zip(evlats, evlons, evdepths,
+                                                     evlabels):
+            arrivals = model.get_ray_paths_geo(
+                    evdepth_km, evlat, evlon, stlat, stlon,
+                    phase_list=phase_list, resample=True)
+            if len(arrivals) == 0:
+                continue
+
+            for arr in arrivals:
+                radii = (r_earth - arr.path['depth']) / r_earth
+                thetas = np.radians(90. - arr.path['lat'])
+                phis = np.radians(arr.path['lon'])
+
+                if coordinate_system == 'RTP':
+                    gcircle = np.array([radii, thetas, phis])
+
+                if coordinate_system == 'XYZ':
+                    gcircle = np.array([radii * np.sin(thetas) * np.cos(phis),
+                                        radii * np.sin(thetas) * np.sin(phis),
+                                        radii * np.cos(thetas)])
+
+                value = 0.
```

Comment:
@MMesch can you elaborate on the significance of always adding 0.0 to the output tuple?

Reply:
Also it seems like bad design to include spaces for padding in the textual headers of the paths.. this should be done in the plotting part.. (coordinates in data array for path omitted below)

```
[(array([[...]]), u'P', 0.0, u'   NET.STA', u'  2017-10-12 | M7.0')]```

Reply:
I'll also include the event id to the output tuple

Reply:
in the long run this routine could also use the information in the `Event.arrivals` to figure out what rays really exist in the data set.. but I guess in many cases such detailed info is not included in the Event data..